src/android/com/microsoftopentechnologies/intellij/wizards/createvm/CloudServiceStep.java

```diff
@@ -0,0 +1,322 @@
+/**
+ * Copyright 2014 Microsoft Open Technologies Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.microsoftopentechnologies.intellij.wizards.createvm;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.ui.components.JBScrollPane;
+import com.intellij.ui.wizard.WizardNavigationState;
+import com.intellij.ui.wizard.WizardStep;
+import com.microsoftopentechnologies.intellij.helpers.UIHelper;
+import com.microsoftopentechnologies.intellij.helpers.azure.AzureCmdException;
+import com.microsoftopentechnologies.intellij.helpers.azure.sdk.AzureSDKManagerImpl;
+import com.microsoftopentechnologies.intellij.model.vm.CloudService;
+import com.microsoftopentechnologies.intellij.model.vm.StorageAccount;
+import com.microsoftopentechnologies.intellij.model.vm.VirtualMachineImage;
+
+import javax.swing.*;
+import javax.swing.event.HyperlinkEvent;
+import javax.swing.event.HyperlinkListener;
+import java.awt.*;
+import java.awt.event.ItemEvent;
+import java.awt.event.ItemListener;
+import java.text.DateFormat;
+import java.util.List;
+import java.util.Vector;
+
+public class CloudServiceStep extends WizardStep<CreateVMWizardModel> {
+
+    private Project project;
+    private CreateVMWizardModel model;
+    private JPanel rootPanel;
+    private JList createVmStepsList;
+    private JEditorPane imageDescriptionTextPane;
+    private JComboBox cloudServiceComboBox;
+    private JComboBox storageComboBox;
+    private JCheckBox availabilitySetCheckBox;
+    private JComboBox availabilityComboBox;
+    private List<StorageAccount> storageAccounts;
+    private List<CloudService> cloudServices;
+
+
+    public CloudServiceStep(CreateVMWizardModel mModel, final Project project) {
+
+        super("Cloud Service Settings");
+
+        this.project = project;
+        this.model = mModel;
+
+        model.configStepList(createVmStepsList, 3);
+
+        cloudServiceComboBox.addItemListener(new ItemListener() {
+            @Override
+            public void itemStateChanged(ItemEvent itemEvent) {
+                fillStorage();
+                fillAvailabilitySets();
+            }
+        });
+
+        cloudServiceComboBox.setRenderer(new DefaultListCellRenderer() {
```

Comment:
An IntelliJ inspection complains that using a `ListCellRenderer` derived class can cause rendering issues on GTK+. Should we consider using `ListCellRendererWrapper` or `ColoredListCellRenderer`?